### PR TITLE
Admin mode: Add admin flag to Zooniverse site header

### PR DESCRIPTION
**Pull request by eatyourgreens**
- Refactor `ZooHeaderContainer` as a functional component, with hooks, in both NextJS apps.
- Add React Testing Library tests that check for the site navigation, test the `signOut` callback and test the Sign In and Register buttons.
- add the `isAdmin` flag to `ZooHeader` on project pages.
- rewrite `ZooHeader `tests using `@testing-library/react` and test `ZooHeader` for admin users in admin mode.
- delete all the Enzyme tests from `ZooHeader/components`.
- delete `ZooHeaderContainer`.
- fix serious accessibility failures on narrow screens. None of the navigation links or menus were properly labelled.

## Package
app-content-pages
app-project
lib-react-components

## How to Review
Project pages now have an 'Admin' link in the Zooniverse site menu when Admin Mode is enabled. Otherwise, the changes here shouldn't change how the site header works.

Screenshot of I Fancy Cats in Admin Mode, with an 'Admin' link in the main Zooniverse site navigation menu.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [x] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [x] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [x] Can submit a classification
- [x] Can sign-in and sign-out
- [x] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
- [x] The PR creator has listed user actions to use when testing the new feature
- [x] Unit tests are included for the new feature
- [x] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

## Refactoring
- [x] The PR creator has described the reason for refactoring
- [x] The refactored component(s) continue to work as expected

### CODE REVIEW

**Comment by eatyourgreens** [comment]
After rewriting the `ZooHeader` tests with React Testing Library, then deleting all the Enzyme tests from sub-folders, the coverage looks pretty good.
Coverage report for ZooHeader, from @zooniverse/react-components, showing 100% coverage for all sub-components.

**Comment by goplayoutside3** [comment]
@mcbouslog I think I want to tag you in as a reviewer for this one because of your familiarity with notifications and sugar. I can't seem to figure out how to test unread notifications.

I ran app-project with this branch locally and visited a project page (PH TESS) while signed in to my account. I triggered 2 notifications on the production site by commenting on a Talk thread. I can see the 2 notifications in ZooHeader while viewing a PFE page such as All Projects on the production site. However, while viewing PH TESS on the locally run app-project, ZooHeader suddenly showed 6 notifications. Then I refreshed the local TESS page and now zero notifications show up.

The behavior I described seemed like a bug possibly on this branch, but then I realized that I when I'm signed in to my account on the production site, and visit an FEM project page such as TESS, I don't see any notifications there either. I can only see unread notifications in ZooHeader on a PFE page. Is it expected to never see unread notifications in FEM's ZooHeader?

**Comment by eatyourgreens** [comment]
There are definitely bugs with unread notifications vanishing, see https://github.com/zooniverse/front-end-monorepo/issues/3672. I’m not sure if those are the same bug or two different bugs.

File: packages/app-content-pages/src/shared/components/ZooHeaderWrapper/ZooHeaderWrapperContainer.js [hunk 1]
```diff
@@ -1,121 +1,63 @@
...
-    display_name: string,
-    login: string,
-    isLoggedIn: bool
-  })
+function ZooHeaderWrapperContainer(props) {
+  const { user } = useStore()
+  const router = useRouter()
+  const onSignOut = useCallback(() => signOut(user), [user])
+  const { display_name, login } = user
+  const userProp = user.isLoggedIn ? { display_name, login } : {}
```
Comment (mcbouslog) [inline packages/app-content-pages/src/shared/components/ZooHeaderWrapper/ZooHeaderWrapperContainer.js:34]: For the lib-react-components ZooHeader MainNavList `isAdmin={user?.admin && isAdmin}`, specifically `user?.admin` to be defined the following suggestion would be needed:

```suggestion
  const { admin, display_name, login } = user
  const userProp = user.isLoggedIn ? { admin, display_name, login } : {}
```

Or delete `user?.admin` from  lib-react-components ZooHeader MainNavList.

**Review by mcbouslog — CHANGES_REQUESTED** [review]
Generally looks great! One small change requested that would either apply to both apps or ZooHeader. 

I really like the consolidation of tests to main component and tests are very well written.

I tested starting from signed out, to singed in, enabling/disabling admin, navigating across project pages, signing out, and sign back in. All was as expected.

**Comment by mcbouslog** [comment]
Regarding bug with Notifications and Messages, which is unrelated to these changes, I think `self.notifications.fetchAndSubscribe()` is missing from the [UserPersonalizaiton store `load` action](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/app-project/stores/User/UserPersonalization/UserPersonalization.js#L71-L74), I'll update a related issue or open a PR asap.

File: packages/lib-react-components/src/ZooHeader/ZooHeader.spec.js [hunk 2]
```diff
@@ -47,11 +51,42 @@ describe('ZooHeader', function () {
...
+    it('should have a user menu button', function () {
+      expect(userMenu).to.exist()
+    })
+  })
+
+  describe('admin user in Admin Mode', function () {
+    let adminLink, userNavigation, notifications, messages, userMenu
+
+    before(function () {
+      const user = { display_name: 'Zoo Admin', login: 'zoo-admin', admin: true }
```
[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/lib-react-components/src/ZooHeader/ZooHeader.spec.js:82]: Good catch.

[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/lib-react-components/src/ZooHeader/ZooHeader.spec.js:82]: I’m puzzled now. How did I get the Admin link to show up in the screenshot at the top of this PR, if I wasn’t passing the `user.admin` flag from the API.

[the remark this answers is not in the document]
Comment (mcbouslog) [reply packages/lib-react-components/src/ZooHeader/ZooHeader.spec.js:82]: Maybe early iteration when `user?.admin` was omitted or something else different? I was puzzled as well, I yarn bootstrapped and rebuilt lib-react-components more times than I care to admit 😆  convince I was missing/doing something wrong.

**Review by mcbouslog — APPROVED** [review]

**Pull request merged**